crates/sui-replay-2/src/lib.rs

```diff
@@ -173,19 +175,27 @@ impl FromStr for Node {
     }
 }
 
-pub async fn handle_replay_config(config: &ReplayConfig, version: &str) -> anyhow::Result<PathBuf> {
-    let ReplayConfig {
-        node,
+pub async fn handle_replay_config(
+    stable_config: &ReplayConfigStable,
+    experimental_config: Option<&ReplayConfigExperimental>,
+    node: &Node,
+    version: &str,
+) -> anyhow::Result<PathBuf> {
+    let ReplayConfigStable {
         digest,
         digests_path,
-        trace,
         mut terminate_early,
+        trace,
         output_dir,
-        show_effects: _,
-        overwrite_existing,
-        verbose,
-        store_mode,
-    } = config;
+        show_effects: _, // used in the caller
+        overwrite: overwrite_existing,
+    } = stable_config;
+
+    let (verbose, store_mode) = if let Some(experimental_config) = experimental_config {
+        (experimental_config.verbose, experimental_config.store_mode)
+    } else {
+        (false, StoreMode::GqlOnly)
+    };
```

Comment:
This seems to imply that there is a reasonable `Default` impl for `ReplayConfigExperimental` -- is it worth defining that, and defining things in terms of that?

In the indexer and RPC codebases, for arguments that can be defaulted, we tend to adopt this pattern:


```
struct Foo {
  #[arg(default_value_t = Default::default().bar)]
  bar: u64, 
}

impl Default for Foo {
  // ... [snip] ...
}
```

I.e. if we can make sure that all the args have defaults (which we try to), we gather them up in a `Default` impl, and refer to that `Default` impl in the `arg` annotations, to avoid duplication.

Reply:
As much as I hate `Default` this is one of the _very_ rare times I can see it being an actually useful thing...

Reply:
Implemented `Default` for `ReplayConfigExperimental` and got rid of `None` experimental setup stuff